src/content/documentation/develop/best-practices-for-collecting-user-data-consents.md

```diff
@@ -152,24 +173,24 @@ You can prompt the user to uninstall the extension with [`management.uninstallSe
 
 ## Your consent dialogs
 
-Once you understand your privacy consent flow, you can add a suitable consent dialog. The following are suggestions on how to present a data collection consent dialog. Where the mockups include the linked text “our Privacy Policy”, this should go to your extension’s privacy policy page on AMO. If you choose to follow these mockups, remember to modify them appropriately for your extension.
+Once you understand your privacy consent flow, you can add a suitable consent dialog. The following are suggestions on how to present a data collection consent dialog. We recommend that all of the data collection controls and actions (toggles and confirmation buttons) be present within the same view. A summary of the data collected and how it is used must be present for each type of data collected within the consent dialog.
+
+Where the mockups include the linked text “our Privacy Policy”, this should go to your extension’s privacy policy page on AMO. If you choose to follow these mockups, remember to modify them appropriately for your extension. 
 
 ### Only personally identifying information
 The default option is not to collect personally identifying information. If you do, the user needs to actively opt-in. Remember to list the data you’re collecting; don’t make the user read your privacy policy to determine what data you are collecting and why.
 
-  ![Mockup of a prompt that could be used when an extension requires consent for processing personal data only.](/assets/img/documentation/develop/privacy_prompt_mockup_personal_data.png)
+  ![Mockup of a prompt that could be used when an extension requires consent for processing personal data only.](/assets/img/documentation/develop/consent-personal.jpg)
 ### Only technical or interaction data
 Data collected does not include user identifiers. The option to collect technical data can be set as the default response.
 
-  ![Mockup of a prompt that could be used when an extension requires consent for processing technical data only.](/assets/img/documentation/develop/privacy_prompt_mockup_anonymous_data.png)
+  ![Mockup of a prompt that could be used when an extension requires consent for processing technical data only.](/assets/img/documentation/develop/consent-anon.jpg)
 ### Combination: Personal and technical data.
 The add-on is requesting both types of data collection. Please ensure the choices are separate.
 
-  ![Mockup of a prompt that could be used when extension requires consent for processing personal and technical data.](/assets/img/documentation/develop/privacy_prompt_mockup_combined.png)
+  ![Mockup of a prompt that could be used when extension requires consent for processing personal and technical data.](/assets/img/documentation/develop/consent-mixed.jpg)
 ### Required data collection
```

Comment:
@wagnerand @abhn If we now recommending the use of a single screen, then doesn't this section need to be rolled into the previous ones? That is, each consent screen could have up to three buttons:

- consent to data collection (and use all of the extensions functions)
- decline data collection, and use the reduced functionality of the extension
- decline data collection and remove the extension (this could either be because it won't work without data collection or simply to allow the user to uninstall it because the reduced functionality is no longer of interest to them)

Reply:
@rebloor if I understood what you're suggesting

```
 -   decline data collection, and use the reduced functionality of the extension
 -   decline data collection and remove the extension (this could either be because it won't work without data collection or simply to allow the user to uninstall it because the reduced functionality is no longer of interest to them)
```

These two will be in an either-or configuration, essentially either the developer allows for usage with reduced functionality, or prompts user to uninstall extension as the primary functionality of the add-on doesn't work without the said data collection. 

I don't see the two buttons being used together, but please feel free to correct me here if you think it is a possible scenario.

Reply:
@abhn We have two cases:
- the app doesn't work without collection data, the user's options are:
  - yes, you can collect data in accordance with your privacy policy (and I will use the app)
  - no, you can't collect data and, as the app won't work without data collection, please delete it for me
- the app works with limited functionality without data collection, user's options are:
  - yes, you can collect data in accordance with your privacy policy (and I will use the app and have all features available)
  - no, you can't collect data and I will use the reduced functionality of the app
  - no, you can't collect data and the reduced functionality of the app is of no use to me; please delete it for me.

Any clearer?

Reply:
@rebloor 

I'm a bit unsure about your suggestion as the consent best practices page is documenting best practices from a developer's perspective and what you're suggesting seems to me like a decision on the user's part. 

Specifically, we let the developer decide if their extension is functional without the said data collection. If it is, they can allow reduced functionality. If not, they'd uninstall the extension on consent decline.

Providing a button that allows the user to keep the extension when the add-on would probably not do anything without said data collection wouldn't make sense, and the developer would be the best judge of that.

```
> no, you can't collect data and I will use the reduced functionality of the app
> no, you can't collect data and the reduced functionality of the app is of no use to me; please delete it for me.
```

Here, the developer decides whether they'd like to keep their add-on usable despite of consent rejection, or if they think it doesn't make sense and uninstall. So either-or between those options.

At least that's how I interpret the following policy statement, derived from which is this choice. 

```
If the main functionality of the add-on does not work without collecting personal data, the add-on must provide a choice for the user to either accept the data collection or uninstall the add-on.
```

Reply:
@abhn I'll leave it there, if you're happy, as I wasn't involved in the policy development. However the statement

> If the main functionality of the add-on does not work without collecting personal data, the add-on must provide a choice for the user to either accept the data collection or uninstall the add-on.

seems somewhat arbitrary in saying that the developer can't offer the non-main functionality that may work without collecting data to a user who doesn't want their data to be collected. (I also have to wonder who determines what constitutes the main functionality.)

I would suggest it should be

> If the add-on does not work without collecting personal data, the add-on must provide a choice for the user to either accept the data collection or uninstall the add-on.

Otherwise the developer and/or user should surely be able to choose if they want a reduced functionality extension?

Reply:
Following our discussion from earlier today, we'll defer the discussion of letting the user decide whether they want to keep the extension with reduced functionality or uninstall it (instead of having the developer decide and reviewers verify the main functionality aspect) to a later time when we're updating the policy itself.

Since we're only updating the best practices documentation here based off of the existing policies, I'll resolve this conversation. Thank you for the feedback in any case. @rebloor